net/discovery.go

```diff
@@ -120,6 +133,11 @@ func (d *discovery) advertise() {
 			d.provides = []types.ProvidesCoin{types.ProvidesXMR}
 			doAdvertise()
 		case <-time.After(ttl):
+			offers := d.offerAPI.GetOffers()
+			if len(offers) == 0 {
+				continue
+			}
+
 			doAdvertise()
```

Comment:
I suspect the average reader, including myself right now, won't know the finer nuances of libp2p DHT caching.  We may need some comments somewhere in here. It makes sense to not advertise if there are no offers, assuming that we don't need to advertise to clear previous offers. Is this an optimization, or is it changing behavior?

This function would be easier to understand if the long nested function was not nested and if this function was called `advertiseLoop`.

Reply:
updated, and added comments explaining things more